pkg/flowaggregator/flowrecord/record.go

```diff
@@ -0,0 +1,496 @@
+		r.IngressNetworkPolicyRuleAction = ingressNPRuleAction.GetUnsigned8Value()
+	}
+	if egressNPName, _, ok := record.GetInfoElementWithValue("egressNetworkPolicyName"); ok {
+		r.EgressNetworkPolicyName = egressNPName.GetStringValue()
+	}
+	if egressNPNamespace, _, ok := record.GetInfoElementWithValue("egressNetworkPolicyNamespace"); ok {
+		r.EgressNetworkPolicyNamespace = egressNPNamespace.GetStringValue()
+	}
+	if egressNPRuleName, _, ok := record.GetInfoElementWithValue("egressNetworkPolicyRuleName"); ok {
+		r.EgressNetworkPolicyRuleName = egressNPRuleName.GetStringValue()
+	}
+	if egressNPType, _, ok := record.GetInfoElementWithValue("egressNetworkPolicyType"); ok {
+		r.EgressNetworkPolicyType = egressNPType.GetUnsigned8Value()
+	}
+	if egressNPRuleAction, _, ok := record.GetInfoElementWithValue("egressNetworkPolicyRuleAction"); ok {
+		r.EgressNetworkPolicyRuleAction = egressNPRuleAction.GetUnsigned8Value()
+	}
+	if tcpState, _, ok := record.GetInfoElementWithValue("tcpState"); ok {
+		r.TcpState = tcpState.GetStringValue()
+	}
+	if flowType, _, ok := record.GetInfoElementWithValue("flowType"); ok {
+		r.FlowType = flowType.GetUnsigned8Value()
+	}
+	if sourcePodLabels, _, ok := record.GetInfoElementWithValue("sourcePodLabels"); ok {
+		r.SourcePodLabels = sourcePodLabels.GetStringValue()
+	}
+	if destinationPodLabels, _, ok := record.GetInfoElementWithValue("destinationPodLabels"); ok {
+		r.DestinationPodLabels = destinationPodLabels.GetStringValue()
+	}
+	if throughput, _, ok := record.GetInfoElementWithValue("throughput"); ok {
+		r.Throughput = throughput.GetUnsigned64Value()
+	}
+	if reverseThroughput, _, ok := record.GetInfoElementWithValue("reverseThroughput"); ok {
+		r.ReverseThroughput = reverseThroughput.GetUnsigned64Value()
+	}
+	if throughputFromSrcNode, _, ok := record.GetInfoElementWithValue("throughputFromSourceNode"); ok {
+		r.ThroughputFromSourceNode = throughputFromSrcNode.GetUnsigned64Value()
+	}
+	if throughputFromDstNode, _, ok := record.GetInfoElementWithValue("throughputFromDestinationNode"); ok {
+		r.ThroughputFromDestinationNode = throughputFromDstNode.GetUnsigned64Value()
+	}
+	if revTputFromSrcNode, _, ok := record.GetInfoElementWithValue("reverseThroughputFromSourceNode"); ok {
+		r.ReverseThroughputFromSourceNode = revTputFromSrcNode.GetUnsigned64Value()
+	}
+	if revTputFromDstNode, _, ok := record.GetInfoElementWithValue("reverseThroughputFromDestinationNode"); ok {
+		r.ReverseThroughputFromDestinationNode = revTputFromDstNode.GetUnsigned64Value()
+	}
+}
+
+// used for unit testing
+func createElement(name string, enterpriseID uint32) ipfixentities.InfoElementWithValue {
+	element, _ := ipfixregistry.GetInfoElement(name, enterpriseID)
+	ieWithValue, _ := ipfixentities.DecodeAndCreateInfoElementWithValue(element, nil)
+	return ieWithValue
+}
+
+// used for unit testing
+func PrepareMockRecord(mockRecord *ipfixentitiestesting.MockRecord, isIPv4 bool) {
```

Comment:
if these are used for testing, they should probably not be in this file
they can go to record_test.go or be placed under `pkg/flowaggregator/flowrecord/testing/`

Reply:
I also feel they should not be there, but I can't come up with a good solution. These functions are shared by UT across three packages, called by record_test.go, s3uploader_test.go, clickhouse_test.go

- If put in to record_test.go, the later two package cannot access them
- If put in `pkg/flowaggregator/flowrecord/testing/`, `GetTestFlowRecord()` returns `*flowrecord.FlowRecord`, record_test.go calls `testing.PrepareMockRecord()`. There will be an import cycle.

Do you have any suggestion?

Reply:
I suggest that you place `PrepareMockRecord` under `pkg/flowaggregator/testing/` and `GetTestFlowRecord` under `pkg/flowaggregator/flowrecord/testing/`